packages/client/src/hooks/useTreasury.js

```diff
@@ -458,6 +493,28 @@ export default function useTreasury(treasuryAddr) {
     await tx(res).onceSealed();
     await refreshTreasury();
   };
+
+  const proposeAddVault = async (coinType) => {
+    const contractName = COIN_TYPE_TO_META[coinType].contractName;
```

Comment:
can we move this line into `try` block too and check COIN_TYPE_TO_META[coinType] exist before accessing contractName? If it does's exist, we can log this coinType is not supported.

Reply:
I can move it to try block. However, the error is not possible to happen because coinType is selected through a dropdown, the only way user can get this error is if they tamper the html, and in that case if they break the app it's only in their browser

Reply:
yea that's right. Not a big problem anyways. (i was thinking more from a function perspective as any client could invoke this method, add safe check to explicitly error out is better than undefined runtime exception from javascript.)